examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/MCCastingPlayer.mm

```diff
@@ -41,53 +45,118 @@ + (NSInteger)kMinCommissioningWindowTimeoutSec
 }
 
-- (void)verifyOrEstablishConnectionWithCompletionBlock:(void (^_Nonnull)(NSError * _Nullable))completion timeout:(long long)timeout desiredEndpointFilter:(MCEndpointFilter * _Nullable)desiredEndpointFilter
+- (NSError *)verifyOrEstablishConnectionWithCallbacks:(MCConnectionCallbacks * _Nonnull)connectionCallbacks
+                     identificationDeclarationOptions:(MCIdentificationDeclarationOptions * _Nullable)identificationDeclarationOptions
 {
-    ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCompletionBlock() called");
-    VerifyOrReturn([[MCCastingApp getSharedInstance] isRunning], ChipLogError(AppServer, "MCCastingApp NOT running"));
+    ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() called, MCConnectionCallbacks and MCIdentificationDeclarationOptions parameters");
+    return [self verifyOrEstablishConnectionWithCallbacks:connectionCallbacks
+                                                  timeout:kMinCommissioningWindowTimeoutSec
+                         identificationDeclarationOptions:identificationDeclarationOptions];
+}
+
+- (NSError *)verifyOrEstablishConnectionWithCallbacks:(MCConnectionCallbacks * _Nonnull)connectionCallbacks
+                                              timeout:(long)timeout
+                     identificationDeclarationOptions:(MCIdentificationDeclarationOptions * _Nullable)identificationDeclarationOptions
+{
+    ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() called, MCConnectionCallbacks, timeout and MCIdentificationDeclarationOptions parameters");
+    VerifyOrReturnValue([[MCCastingApp getSharedInstance] isRunning],
+        [MCErrorUtils NSErrorFromChipError:CHIP_ERROR_INCORRECT_STATE],
+        ChipLogError(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() MCCastingApp NOT running"));
 
     dispatch_queue_t workQueue = [[MCCastingApp getSharedInstance] getWorkQueue];
     dispatch_sync(workQueue, ^{
-        matter::casting::core::IdentificationDeclarationOptions idOptions;
-
-        // TODO: In the following PRs. Replace EndpointFilter objC class with IdentificationDeclarationOptions objC class.
-        __block matter::casting::core::EndpointFilter cppDesiredEndpointFilter;
-        if (desiredEndpointFilter != nil) {
-            chip::Protocols::UserDirectedCommissioning::TargetAppInfo targetAppInfo;
-            targetAppInfo.vendorId = desiredEndpointFilter.vendorId;
-            targetAppInfo.productId = desiredEndpointFilter.productId;
-
-            CHIP_ERROR result = idOptions.addTargetAppInfo(targetAppInfo);
-            if (result != CHIP_NO_ERROR) {
-                ChipLogError(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCompletionBlock() failed to add targetAppInfo: %" CHIP_ERROR_FORMAT, result.Format());
-            }
+        matter::casting::core::IdentificationDeclarationOptions cppIdOptions;
```

Comment:
This method implementation is getting pretty long. Can you move the setup of the cppIdOptions to a separate internal (private) method in MCCastingPlayer (do not expose publicly)?

Reply:
Adding setupCppIdOptions method defined as a private method within the @interface section. In Objective-C, methods defined within a class's @implementation section are effectively private if they are not declared in the public header file.
However "pretty long" is arbitrary, we should define a concrete standard.